function-aws-api-proxy/src/test/groovy/io/micronaut/function/aws/proxy/MicronautLambdaHandlerSpec.groovy

```diff
@@ -109,6 +110,95 @@ class MicronautLambdaHandlerSpec extends Specification {
         handler.close()
     }
 
+    @Issue("https://github.com/micronaut-projects/micronaut-aws/issues/1410")
+    void "POST form url encoded body binding to pojo works"() {
+        given:
+        MicronautLambdaContainerHandler handler = instantiateHandler()
+
+        when:
+        AwsProxyRequestBuilder builder = new AwsProxyRequestBuilder('/form', HttpMethod.POST.toString())
+        builder.header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_FORM_URLENCODED)
+        builder.body("message=World")
+        AwsProxyResponse response = handler.proxy(builder.build(), new MockLambdaContext())
+
+        then:
+        response.statusCode == 200
+        response.body == '{"message":"Hello World"}'
+    }
+
+    @PendingFeature
+    void "POST form url encoded body binding to pojo works if you don't specify body annotation"() {
+        given:
+        MicronautLambdaContainerHandler handler = instantiateHandler()
+
+        when:
+        AwsProxyRequestBuilder builder = new AwsProxyRequestBuilder('/form/without-body-annotation', HttpMethod.POST.toString())
+        builder.header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_FORM_URLENCODED)
+        builder.body("message=World")
+        AwsProxyResponse response = handler.proxy(builder.build(), new MockLambdaContext())
+
+        then:
+        response.statusCode == 200
+        response.body == '{"message":"Hello World"}'
+    }
+
+    @Issue("https://github.com/micronaut-projects/micronaut-aws/issues/1410")
+    void "form-url-encoded with Body annotation and a nested attribute"() {
+        given:
+        MicronautLambdaContainerHandler handler = instantiateHandler()
+
+        when:
+        AwsProxyRequestBuilder builder = new AwsProxyRequestBuilder('/form/nested-attribute', HttpMethod.POST.toString())
+        builder.header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_FORM_URLENCODED)
+        builder.body("message=World")
+        AwsProxyResponse response = handler.proxy(builder.build(), new MockLambdaContext())
+
+        then:
+        response.statusCode == 200
+        response.body == '{"message":"Hello World"}'
+
+        cleanup:
+        handler.close()
+    }
+
+    @Issue("https://github.com/micronaut-projects/micronaut-aws/issues/1410")
+    void "application-json with Body annotation and a nested attribute"() {
+        given:
+        MicronautLambdaContainerHandler handler = instantiateHandler()
+
+        when:
+        AwsProxyRequestBuilder builder = new AwsProxyRequestBuilder('/form/json-nested-attribute', HttpMethod.POST.toString())
+        builder.header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON)
+        builder.body("{\"message\":\"World\"}")
+        AwsProxyResponse response = handler.proxy(builder.build(), new MockLambdaContext())
+
+        then:
+        response.statusCode == 200
+        response.body == '{"message":"Hello World"}'
+
+        cleanup:
+        handler.close()
+    }
+
+    @PendingFeature
```

Comment:
returning a map and having the map rendered as JSON is something that works with JSON